Packs/InfobloxBloxOne/Integrations/InfobloxBloxOneThreatDefenseEventCollector/InfobloxBloxOneThreatDefenseEventCollector.py

```diff
@@ -0,0 +1,104 @@
+from CommonServerPython import *
+from CommonServerUserPython import *
+import demistomock as demisto
+
+
+VENDOR = 'Infoblox BloxOne'
+PRODUCT = 'Threat Defense'
+
+
+class BloxOneTDEventCollectorClient(BaseClient):
+    def __init__(self, api_key: str, verify=True, proxy=False):
+        super(BloxOneTDEventCollectorClient, self).__init__(
+            headers={'Authorization': f'Token {api_key}'},
+            base_url='https://csp.infoblox.com',
+            verify=verify,
+            proxy=proxy
+        )
+
+    def fetch_events(self, from_ts: int, to_ts: int, limit: int = 1000, offset: int = 0) -> List[Dict]:
+        def map_time(event: Dict) -> Dict:
+            event['_time'] = event['event_time']
+            return event
+
+        events = self._http_request('GET', '/api/dnsdata/v2/dns_event',
+                                    params={'t0': from_ts, 't1': to_ts, '_limit': limit, '_offset': offset}
+                                    )['result']
+        return list(map(map_time, events))
+
+
+def fetch_events_command(client: BloxOneTDEventCollectorClient, params: Dict, last_run: Dict):
+    from_ts = last_run.get('from_ts')
+    if from_ts is None:
+        from_date_time = dateparser.parse(params.get('first_fetch', '1 day'), settings={'TIMEZONE': 'UTC'})
+        if from_date_time is None:
+            raise DemistoException('Invalid date format in "First fetch time interval" parameter')
+
+        from_ts = int(from_date_time.timestamp())
+
+    current_ts = int(datetime.utcnow().timestamp())
```

Comment:
We don't want to take the last event's timestamp to be the `current_ts`?

Reply:
the current_ts is used for the `to` (AKA `t1`) argument as a part of the API itself (and not just for the next run)